libvast/src/system/application.cpp

```diff
@@ -49,4 +36,28 @@ application::application() {
   detail::adjust_resource_consumption();
 }
 
+void render_error(const application& app, const caf::error& err,
+                  std::ostream& os) {
+  if (err)
+    // The user most likely killed the process via CTRL+C, print nothing.
+    return;
+  os << render(err);
+  if (err.category() == caf::atom("vast")) {
+    auto x = static_cast<vast::ec>(err.code());
+    switch (x) {
+      default:
+        break;
+      case ec::invalid_subcommand:
+      case ec::missing_subcommand:
+      case ec::unrecognized_option:
+        auto ctx = err.context();
```

Comment:
Cases are jump labels, so we should add an extra scope around this block, as it uses local variables. As-is, the code is going to break when adding additional non-empty cases.